app/controllers/raw_file_sources_controller.rb

```diff
@@ -0,0 +1,83 @@
+class RawFileSourcesController < ApplicationController
+	include Alertable
+
+	before_action :authenticate_user! 
+	before_action :set_source, only: [:show, :edit, :update, :destroy]
+
+	def index
+		@raw_file_sources = RawFileSource.all.order(:build_order)
+
+		respond_to do |format|
+			format.html
+		end
+	end
+
+	def new
+		@raw_file_source = RawFileSource.new
+		
+		respond_to do |format|
+			format.html
+		end	
+	end
+
+	def show
+		respond_to do |format|
+			format.html
+		end	
+	end
+
+	def create
+		@raw_file_source = RawFileSource.new(source_params)
+		saved = @raw_file_source.save
+
+		respond_to do |format|
+			if saved
+				format.html { redirect_to @raw_file_source, notice: "#{@raw_file_source.name} saved."}
+			else
+      	label = "Errors prohibited this source from being saved:"
+      	errors = @raw_file_source.errors.full_messages
+				flash.alert = RawFileSourcesController.pretty_error(label, errors).html_safe
+
+        format.html { render :new }
+ 			end
+		end
+	end
+
+	def edit
+		respond_to do |format|
+			format.html
+		end	
+	end
+
+	def update
+    respond_to do |format|
```

Comment:
Nitpick: Formatting seems off here. Is the linter running? (I think it'll auto correct this)